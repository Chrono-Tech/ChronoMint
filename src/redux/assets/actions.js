import contractsManagerDAO from 'dao/ContractsManagerDAO'

export const ASSETS_LIST = 'assets/LIST'

export const listAssets = () => async (dispatch) => {
  const dao = await contractsManagerDAO.getAssetsManagerDAO()
  dispatch({type: ASSETS_LIST, list: await dao.getList()})
}
