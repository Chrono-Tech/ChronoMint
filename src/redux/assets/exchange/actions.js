import type ExchangeManagerDAO from 'dao/ExchangeManagerDAO'
import type ExchangeModel from 'models/ExchangeModel'
import type TokenModel from 'models/TokenModel'
import contractsManagerDAO from 'dao/ContractsManagerDAO'

export const EXCHANGE_MANAGER_LIST = 'exchangeManager/LIST'
export const EXCHANGE_MANAGER_SET = 'exchangeManager/SET'
export const EXCHANGE_MANAGER_REMOVE = 'exchangeManager/REMOVE'

const getDAO = async (): Promise<ExchangeManagerDAO> => await contractsManagerDAO.getExchangeManagerDAO()

export const listExchanges = () => async (dispatch) => {
  const dao = await getDAO()
  dispatch({type: EXCHANGE_MANAGER_LIST, list: await dao.getList()})
}

export const addExchange = (address) => async () => {
  const exchange = new ExchangeModel({address, isFetching: true})
  dispatch({type: EXCHANGE_MANAGER_SET, exchange})
  try {
    const dao = await getDAO()
    await dao.addExchange(address)
  } catch (e) {
    // no rollback
  }
}

export const createExchange = (token: TokenModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.createExchange(token)
  } catch (e) {
    // no rollback
  }
}

export const editExchange = (exchange: ExchangeModel, newExchange: ExchangeModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.editExchange(exchange, newExchange)
  } catch (e) {
    // no rollback
  }
}

export const removeExchange = (exchange: ExchangeModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.removeExchange(exchange)
  } catch (e) {
    // no rollback
  }
}

// TODO @bshevchenko: MINT-230 AssetsManager
// export const addOwner = (exchange: ExchangeModel, account) => async () => {
//   try {
//     const dao = await getDAO()
//     await dao.addOwner(exchange, account)
//   } catch (e) {
//     // no rollback
//   }
// }
//
// export const removeOwner = (exchange: ExchangeModel, account) => async () => {
//   try {
//     const dao = await getDAO()
//     await dao.removeOwner(exchange, account)
//   } catch (e) {
//     // no rollback
//   }
// }
