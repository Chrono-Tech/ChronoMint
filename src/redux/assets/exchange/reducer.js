import Immutable from 'immutable'
import * as a from './actions'

const initialState = {
  list: new Immutable.Map(),
  isFetched: false
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case a.EXCHANGE_MANAGER_LIST:
      return {
        ...state,
        list: action.list,
        isFetched: true
      }
    default:
      return state
  }
}

export default reducer
