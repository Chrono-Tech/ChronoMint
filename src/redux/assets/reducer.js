import Immutable from 'immutable'
import * as a from './actions'

const initialState = {
  list: new Immutable.Map(),
  isFetched: false
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case a.ASSETS_LIST:
      return {
        ...state,
        list: action.list,
        isFetched: true
      }
    default:
      return state
  }
}

export default reducer
