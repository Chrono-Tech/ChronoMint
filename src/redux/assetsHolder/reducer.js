import TimeHolderModel from 'models/timeHolder/TimeHolderModel'
import * as a from './actions'

const initialState = new TimeHolderModel()

export default (state = initialState, action) => {
  switch (action.type) {
    case a.TIME_HOLDER_INIT:
      return state.isInited(true)
    case a.TIME_HOLDER_TIME_ADDRESS:
      return state.timeAddress(action.address)
    case a.TIME_HOLDER_ADDRESS:
      return state.timeHolderAddress(action.address)
    case a.TIME_HOLDER_WALLET_ADDRESS:
      return state.timeHolderWalletAddress(action.address)
    default:
      return state
  }
}
