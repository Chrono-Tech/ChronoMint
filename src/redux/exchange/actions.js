// TODO @bshevchenko: this is intermediate version for demo
import Immutable from 'immutable'
import BigNumber from 'bignumber.js'
import { ETH, LHT } from 'redux/wallet/actions'
import ExchangeOrderModel from 'models/ExchangeOrderModel'
import exchangeDAO from 'dao/ExchangeDAO'
import ethereumDAO from 'dao/EthereumDAO'

export const EXCHANGE_ORDERS = 'exchange/ORDERS'

export const search = (symbol: string, isBuy: boolean) => async dispatch => {
  const isSell = !isBuy

  let limitPromise
  let accountBalancePromise

  if ((symbol === ETH && isBuy) || (symbol === LHT && isSell)) {
    symbol = ETH
    limitPromise = exchangeDAO.getETHBalance()
    accountBalancePromise = exchangeDAO.getAccountAssetBalance()
  } else if ((symbol === ETH && isSell) || (symbol === LHT && isBuy)) {
    symbol = LHT
    limitPromise = exchangeDAO.getAssetBalance()
    accountBalancePromise = ethereumDAO.getAccountBalance()
  }

  const [limit, buyPrice, sellPrice, accountBalance] = await Promise.all([
    limitPromise,
    exchangeDAO.getBuyPrice(),
    exchangeDAO.getSellPrice(),
    accountBalancePromise,
  ])

  const order = new ExchangeOrderModel({
    limit,
    description: 'Auto Exchanger',
    symbol,
    isBuy,
    buyPrice,
    sellPrice,
    accountBalance,
  })

  dispatch({ type: EXCHANGE_ORDERS, orders: new Immutable.List([order]) })
}

export const exchange = (order: ExchangeOrderModel, amount: BigNumber) => async () => {
  try {
    if (order.isBuy()) {
      await exchangeDAO.buy(amount, order.sellPrice())
    } else {
      await exchangeDAO.sell(amount, order.buyPrice())
    }
  } catch (e) {
    // no rollback
  }
}
