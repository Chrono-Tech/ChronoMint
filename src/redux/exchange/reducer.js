import ExchangeModel from 'models/exchange/ExchangeModel'
import * as a from './actions'

const initialState = new ExchangeModel()

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case a.EXCHANGE_GET_ORDERS_START:
      return state.exchanges(state.exchanges().isFetched(false).isFetching(true))
    case a.EXCHANGE_GET_ORDERS_FINISH:
      return state.exchanges(action.exchanges.isFetched(true).isFetching(false)).lastPages(action.lastPages || state.lastPages)
    case a.EXCHANGE_SET_FILTER:
      return state.filter(action.filter)
    case a.EXCHANGE_GET_DATA_START:
      return state.isFetched(false).isFetching(true)
    case a.EXCHANGE_GET_DATA_FINISH:
      return state
        .assetSymbols(action.assetSymbols || state.assetSymbols())
        .isFetched(true).isFetching(false)
    case a.EXCHANGE_MIDDLEWARE_DISCONNECTED:
      return state
        .showFilter(false)
        .isFetched(true).isFetching(false)
    case a.EXCHANGE_GET_TOKENS_LIST_START:
      return state
        .tokens(state.tokens().isFetching(true))
    case a.EXCHANGE_GET_TOKENS_LIST_FINISH:
      return state
        .tokens(action.tokens.isFetched(true).isFetching(false))
    case a.EXCHANGE_REMOVE:
      return state.exchangesForOwner(state.exchangesForOwner().remove(action.exchange))
    case a.EXCHANGE_UPDATE:
      return state.exchangesForOwner(state.exchangesForOwner().update(action.exchange))
    case a.EXCHANGE_EXCHANGES_LIST_GETTING_START:
      return state.exchanges(state.exchanges().isFetching(true))
    case a.EXCHANGE_EXCHANGES_LIST_GETTING_FINISH:
      return state
        .exchanges(state.exchanges().concat(action.exchanges).isFetching(false).isFetched(true))
        .lastPages(action.lastPages)
    case a.EXCHANGE_SET_PAGES_COUNT:
      return state.pagesCount(action.count)
    case a.EXCHANGE_GET_OWNERS_EXCHANGES_START:
      return state.exchangesForOwner(state.exchangesForOwner().isFetched(false).isFetching(true))
    case a.EXCHANGE_GET_OWNERS_EXCHANGES_FINISH:
      return state.exchangesForOwner(action.exchanges.isFetched(true).isFetching(false))
    default:
      return state
  }
}

export default reducer
