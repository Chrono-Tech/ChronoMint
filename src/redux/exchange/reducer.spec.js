import Immutable from 'immutable'
import TokenModel from 'models/tokens/TokenModel'
import ExchangeOrderModel from 'models/exchange/ExchangeOrderModel'
import ExchangesCollection from 'models/exchange/ExchangesCollection'
import TokensCollection from 'models/tokens/TokensCollection'
import * as a from './actions'
import reducer, { initialState } from './reducer'

describe('assetsManager reducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })

  it('sould handle EXCHANGE_GET_ORDERS_START', () => {
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_ORDERS_START,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_GET_ORDERS_FINISH', () => {
    const exchange = new ExchangeOrderModel({})
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_ORDERS_FINISH,
        exchanges: new ExchangesCollection().add(exchange),
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_SET_FILTER', () => {
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_SET_FILTER,
        filter: new Immutable.Map({
          symbol: 'TIME',
        }),
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_GET_DATA_START', () => {
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_DATA_START,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_GET_DATA_FINISH', () => {
    const assetSymbols = [ 'TIME' ]
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_DATA_FINISH,
        assetSymbols,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_MIDDLEWARE_DISCONNECTED', () => {
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_MIDDLEWARE_DISCONNECTED,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_GET_TOKENS_LIST_START', () => {
    const token = new TokenModel({})
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_TOKENS_LIST_START,
        tokens: new TokensCollection().add(token),
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_GET_TOKENS_LIST_FINISH', () => {
    const token = new TokenModel({})
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_TOKENS_LIST_FINISH,
        tokens: new TokensCollection().add(token),
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_REMOVE_FOR_OWNER', () => {
    const exchange = new ExchangeOrderModel({ address: 'test' })
    const state = initialState.exchangesForOwner(new ExchangesCollection().add(exchange))
    expect(
      reducer(state, {
        type: a.EXCHANGE_REMOVE_FOR_OWNER,
        exchange: exchange,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_UPDATE_FOR_OWNER', () => {
    const exchange = new ExchangeOrderModel({ address: 'test' })
    const state = initialState.exchangesForOwner(new ExchangesCollection().add(exchange))
    expect(
      reducer(state, {
        type: a.EXCHANGE_REMOVE_FOR_OWNER,
        exchange: exchange,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_UPDATE', () => {
    const exchange = new ExchangeOrderModel({ address: 'test' })
    const state = initialState.exchangesForOwner(new ExchangesCollection().add(exchange))
    expect(
      reducer(state, {
        type: a.EXCHANGE_UPDATE,
        exchange: exchange,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_EXCHANGES_LIST_GETTING_START', () => {
    const exchange = new ExchangeOrderModel({ address: 'test' })
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_EXCHANGES_LIST_GETTING_START,
        exchanges: new ExchangesCollection().add(exchange),
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_EXCHANGES_LIST_GETTING_FINISH', () => {
    const exchange = new ExchangeOrderModel({ address: 'test' })
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_EXCHANGES_LIST_GETTING_FINISH,
        exchanges: new ExchangesCollection().add(exchange),
        lastPages: 1,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_SET_PAGES_COUNT', () => {
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_SET_PAGES_COUNT,
        count: 1,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_GET_OWNERS_EXCHANGES_START', () => {
    const exchange = new ExchangeOrderModel({ address: 'test' })
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_OWNERS_EXCHANGES_START,
        exchangesForOwner: new ExchangesCollection().add(exchange),
      }),
    ).toMatchSnapshot()
  })

  it('sould handle EXCHANGE_GET_OWNERS_EXCHANGES_FINISH', () => {
    const exchange = new ExchangeOrderModel({ address: 'test' })
    expect(
      reducer(undefined, {
        type: a.EXCHANGE_GET_OWNERS_EXCHANGES_FINISH,
        exchanges: new ExchangesCollection(),
      }),
    ).toMatchSnapshot()
  })
})
