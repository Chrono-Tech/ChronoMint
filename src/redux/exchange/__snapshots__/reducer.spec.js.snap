// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assetsManager reducer sould handle EXCHANGE_EXCHANGES_LIST_GETTING_FINISH 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": true,
    "isFetching": false,
    "isPending": false,
    "list": {
      "test": {
        "address": test,
        "assetBalance": 0,
        "authorizedManager": null,
        "buyPrice": 0,
        "ethBalance": 0,
        "id": null,
        "isActive": true,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isPending": false,
        "owner": null,
        "sellPrice": 0,
        "symbol": null,
        "timestamp": null,
        "transactionHash": null,
      },
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 1,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_EXCHANGES_LIST_GETTING_START 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": true,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_DATA_FINISH 1`] = `
  "assetSymbols": {
    "0": TIME,
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": true,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_DATA_START 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": true,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_ORDERS_FINISH 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": true,
    "isFetching": false,
    "isPending": false,
    "list": {
      "null": {
        "address": null,
        "assetBalance": 0,
        "authorizedManager": null,
        "buyPrice": 0,
        "ethBalance": 0,
        "id": null,
        "isActive": true,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isPending": false,
        "owner": null,
        "sellPrice": 0,
        "symbol": null,
        "timestamp": null,
        "transactionHash": null,
      },
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": Function,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_ORDERS_START 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": true,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_OWNERS_EXCHANGES_FINISH 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": true,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_OWNERS_EXCHANGES_START 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": true,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_TOKENS_LIST_FINISH 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": true,
    "isFetching": false,
    "isPending": false,
    "list": {
      "null": {
        "address": null,
        "allowance": {
        },
        "balance": 0,
        "dao": null,
        "decimals": null,
        "fee": null,
        "feeAddress": null,
        "icon": null,
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isOptional": true,
        "isPending": false,
        "isReissuable": false,
        "managersList": null,
        "name": null,
        "platform": null,
        "symbol": null,
        "timestamp": null,
        "totalSupply": 0,
        "transactionHash": null,
        "url": null,
        "withFee": false,
      },
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_GET_TOKENS_LIST_START 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": true,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_MIDDLEWARE_DISCONNECTED 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": true,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": false,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_REMOVE_FOR_OWNER 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_SET_FILTER 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
    "symbol": TIME,
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_SET_PAGES_COUNT 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 1,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_UPDATE 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
      "test": {
        "address": test,
        "assetBalance": 0,
        "authorizedManager": null,
        "buyPrice": 0,
        "ethBalance": 0,
        "id": null,
        "isActive": true,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isPending": false,
        "owner": null,
        "sellPrice": 0,
        "symbol": null,
        "timestamp": null,
        "transactionHash": null,
      },
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
      "test": {
        "address": test,
        "assetBalance": 0,
        "authorizedManager": null,
        "buyPrice": 0,
        "ethBalance": 0,
        "id": null,
        "isActive": true,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isPending": false,
        "owner": null,
        "sellPrice": 0,
        "symbol": null,
        "timestamp": null,
        "transactionHash": null,
      },
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;

exports[`assetsManager reducer sould handle EXCHANGE_UPDATE_FOR_OWNER 1`] = `
  "assetSymbols": {
  },
  "exchanges": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "exchangesForOwner": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "filter": {
  },
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isPending": false,
  "lastPages": 0,
  "pagesCount": 0,
  "showFilter": true,
  "timestamp": null,
  "tokens": {
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isPending": false,
    "list": {
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null,
  },
  "transactionHash": null,

`;
